src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -1009,6 +1009,16 @@ public void visitSources(SCMSourceObserver observer) throws IOException, Interru
                             System.currentTimeMillis(),
                             String.format(
                                 "Skipping repository %s because it is public", repo.getName())));
+              } else if (repo.getSource() != null
+                  && gitHubSCMNavigatorContext.isExcludeForkedRepositories()) {
```

Comment:
This is a less than ideal behavior that is part of github-api - the API doesn't make it clear which method calls can result in additional REST API calls.  I'm pretty sure some endpoints that return repository info don't include the `source` info.  That means that the `getSource()` method has to check for `null` and make a request if it is `null`.  

In order to prevent an addition REST call we need to switch the order on these conditions. 
```suggestion
              } else if (gitHubSCMNavigatorContext.isExcludeForkedRepositories() 
                  && repo.getSource() != null) {
```

Reply:
oh awesome. I can try and make that change when i have a minute (Would just commit your suggestion but I know its like 10 times in that file)